gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java

```diff
@@ -684,6 +743,187 @@ public Boolean call() {
     }
   }
 
+  @Override
+  public Address get(final AddressId addressId, AddressOption... options) {
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Address answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Address>() {
+            @Override
+            public com.google.api.services.compute.model.Address call() {
+              switch (addressId.type()) {
+                case REGION:
+                  RegionAddressId regionAddressId = (RegionAddressId) addressId;
+                  return computeRpc.getRegionAddress(regionAddressId.region(),
+                      regionAddressId.address(), optionsMap);
+                case GLOBAL:
+                  return computeRpc.getGlobalAddress(addressId.address(), optionsMap);
+                default:
+                  throw new IllegalArgumentException("Unexpected address identity type");
```

Comment:
This is added just to be strict and facilitate the detection of errors. In reality this can and should never happen. I don't think this should be documented.